tests/integration/collection__indexed_element__waiting_search_on_actions_like_click_test.py

```diff
@@ -101,12 +108,14 @@ def test_fails_on_timeout_during_waits_for_present_in_dom_and_visibility(
     page = GivenPage(browser.driver)
     page.opened_with_body(
         '''
-        <h2 id="second">Heading 2</h2>'''
+        <h2 id="second">Heading 2</h2>
+        '''
     )
     page.load_body_with_timeout(
         '''
         <a href="#second">go to Heading 2</a>
-        <h2 id="second">Heading 2</h2>''',
+        <h2 id="second">Heading 2</h2>
+        ''',
         0.5,
     )
 
```

Comment:
below we have an interesting case...
the `pytest.raises` style actually does not allow to separate Assert from Act in the correct order... 
in fact we have AssertException, then Act, then AssertUrl...

that's why i often like more the following, yeah – pretty stupid – but also more KISS and straightforward style:
```python
    try:
        browser.all('a')[0].click()
        pytest.fail("click should not pass")
    
    except TimeoutException:
        assert "second" not in session_browser.driver.current_url
```
 
 here we clearly separate Assert from Act in their natural order
 
 having some drawback – one more line of code with `pytest.fail("click should not pass")` – yet this is might be a good thing;)